<launch>
	<param name="use_sim_time" value="true"/>

	<include file="$(find robowork_gazebo)/launch/bvr_SIM_playpen.launch"/>

	<arg name="robot_namespace" default="bvr_SIM"/>
	<arg name="arm_namespace" default="main_arm_SIM"/> 
	<arg name="sim_suffix" default="_SIM"/>
	<arg name="VIO_enabled" default="true"/>

	<group ns="bvr_SIM">
		<include file="$(find robowork_moveit_config)/launch/robowork_moveit_planning_execution.launch">
			<arg name="robot_namespace" value="$(arg robot_namespace)" />
      <arg name="arm_namespace" value="$(arg arm_namespace)" />
      <arg name="sim_suffix" value="$(arg sim_suffix)" />
		</include>

		<!-- <include file="$(find robowork_ur_launch)/launch/ur5e_compliance.launch">
			<arg name="robot_namespace" value="bvr_SIM"/>
			<arg name="arm_namespace" value="main_arm_SIM"/>
		</include> -->

    <include file="$(find robowork_moveit_config)/launch/planning_context.launch">
      <arg name="robot_namespace" value="$(arg robot_namespace)" />
      <arg name="arm_namespace" value="$(arg arm_namespace)" />
      <arg name="sim_suffix" value="$(arg sim_suffix)" />
      <arg name="VIO_enabled" value="$(arg VIO_enabled)" />
    </include>

    <!-- Planning Functionality -->
    <include ns="move_group" file="$(find robowork_moveit_config)/launch/planning_pipeline.launch.xml">
      <arg name="sim_suffix" value="$(arg sim_suffix)" />
      <arg name="VIO_enabled" value="$(arg VIO_enabled)" />
      <arg name="pipeline" value="ompl" />
    </include>
		
	</group>

	<include file="$(find mm_coverage_planner)/launch/rviz.launch"/>
</launch>